docs/superpowers/specs/2026-07-17-model-first-settings-redesign.md

```
@@ -0,0 +1,933 @@
+- Save validation 성공: key를 저장하고 `API Key configured`
+- validation 실패: 현재 validation error를 표시하고 기존 persisted key를 변경하지 않음
+- Settings 재진입: persisted key가 있으면 `API Key configured`
+- Clear: persisted key와 configured status를 제거해 빈 secure field 상태로 돌아감
+
+`API Key configured`는 Save validation을 통과한 key가 저장되어 있음을 뜻한다. Settings 진입마다 연결을 재확인하지 않으므로 `Connected`, `Valid`, `Available`처럼 지속적인 network health를 보장하는 표현을 사용하지 않는다.
+
+현재 Cloud-only인 Post-processing과 Context에서는 common key가 비어 있으면 switch와 Cloud model selector를 비활성화한다. persisted switch/model 값은 변경하거나 숨기지 않는다. 비활성 control 바로 아래에 icon과 text를 함께 사용한 inline readiness 이유를 표시한다. `Details` 전체는 비활성화하지 않는다.
+
+### Future work — Local AI Server (not implemented in this UI pass)
+
+#64가 #145 Phase 2를 충족하도록 Post-processing과 Context에서 외부 OpenAI-compatible Local AI Server를 선택할 수 있게 한다.
```

markdownlint-cli2 (0.23.0)

[warning] No space after hash on atx style heading

(MD018, no-missing-space-atx)